### PR TITLE
feat: show alert before closing transaction flow

### PR DESCRIPTION
**Pull request by iamacook**
## What it solves

Resolves direct exit of new transaction flow

## How this PR fixes it

A confirmation dialog is now shown when exiting or navigating away from the new transaction flow which needs to first be accepted.

## How to test it

Create a transaction, attempt to close the flow/navigate away with back button and observe the new confirmation dialog.

## Screenshots

image

alert

## Checklist
* [ ] I've tested the branch on mobile 📱
* [ ] I've documented how it affects the analytics (if at all) 📊
* [ ] I've written a unit/e2e test for it (if applicable) 🧑‍💻

### CODE REVIEW

File: src/components/tx-flow/common/TxFlowExitWarning/index.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            color="error"
+            sx={{ verticalAlign: 'middle', mr: '12px' }}
+          />{' '}
+          Are you sure?
+        </>
+      }
+    >
+      <DialogContent sx={{ p: 'var(--space-3) !important' }}>
+        <DialogContentText color={isDarkMode ? undefined : 'primary'}>
+          If you leave this page, the current progress will be lost.
```
Comment (katspaugh) [inline src/components/tx-flow/common/TxFlowExitWarning/index.tsx:37]: This sounds like a threat.

File: src/components/tx-flow/flows/SignMessage/SignMessage.tsx [hunk 2]
```diff
@@ -175,9 +130,9 @@ export type ConfirmProps = BaseProps & {
   requestId?: RequestId
 }
 
-const SignMessage = ({ onClose, message, safeAppId, requestId }: ProposeProps | ConfirmProps): ReactElement => {
+const SignMessage = ({ message, safeAppId, requestId }: ProposeProps | ConfirmProps): ReactElement => {
   // Hooks & variables
-  const [showCloseTooltip, setShowCloseTooltip] = useState<boolean>(false)
+  const { onClose } = useContext(TxModalContext)
```
Comment (katspaugh) [inline src/components/tx-flow/flows/SignMessage/SignMessage.tsx:135]: How is this used? It was supposed to be an internal callback that outside components should only assign, but not call. Only TxModalProvider should call it. It shouldn't even be exported (my mistake that it is).

Comment (iamacook) [reply src/components/tx-flow/flows/SignMessage/SignMessage.tsx:135]: When the user submits the [last sync signature](https://github.com/safe-global/safe-wallet-web/blob/dev/src/hooks/messages/useSyncSafeMessageSigner.ts#L37-L57) or it is received [via poll](https://github.com/safe-global/safe-wallet-web/blob/dev/src/hooks/messages/useSyncSafeMessageSigner.ts#L21-L25), it [closes the modal](https://github.com/safe-global/safe-wallet-web/blob/dev/src/services/safe-messages/safeMsgNotifications.ts#L34-L40).

Comment (katspaugh) [reply src/components/tx-flow/flows/SignMessage/SignMessage.tsx:135]: In this scenario you’re supposed to call setTxFlow(undefined).

Comment (iamacook) [reply src/components/tx-flow/flows/SignMessage/SignMessage.tsx]: Good point. I've updated it accordingly, as well as removing `onClose` from the context.

**Review by katspaugh — APPROVED** [review]

**Pull request merged**